src/alignment/mod.rs

```diff
@@ -229,6 +231,8 @@ mod tests {
             score: 5,
             xstart: 3,
             ystart: 0,
+            xend: 10,
+            yend: 10,
```

Comment:
xlen is used to get the clipping entries in the cigar string.  xend and yend are redundant given the operations vector, but you have to loop over that vector to reconstruct.  so they could be implemented as methods if you'd prefer.